electron-app/tests/unit/components/display/amount-display.spec.ts

```diff
@@ -5,48 +5,102 @@ import Vuetify from 'vuetify';
+function createWrapper(
+  value: BigNumber,
+  amount: BigNumber,
+  fiatCurrency: string | null
+) {
   let vuetify: typeof Vuetify = new Vuetify();
   return mount(AmountDisplay, {
     store,
     vuetify,
     stubs: ['v-tooltip'],
     propsData: {
       value,
-      fiat
+      fiatCurrency,
+      amount
     }
   });
 }
 
 describe('AmountDisplay.vue', () => {
   let wrapper: Wrapper<AmountDisplay>;
 
-  beforeEach(() => {
-    store.commit('session/defaultCurrency', currencies[1]);
-    store.commit('balances/usdToFiatExchangeRates', { EUR: 1.2 });
-    store.commit('session/settings', { floatingPrecision: 2 });
-  });
+  describe('Common case', () => {
+    beforeEach(() => {
+      store.commit('session/defaultCurrency', currencies[1]);
+      store.commit('balances/usdToFiatExchangeRates', { EUR: 1.2 });
+      store.commit('session/settings', { floatingPrecision: 2 });
+    });
 
-  afterEach(() => {
-    store.commit('session/reset');
-  });
+    afterEach(() => {
+      store.commit('session/reset');
+    });
+
+    test('displays amount converted to selected fiat currency', async () => {
+      wrapper = createWrapper(bigNumberify(1.20440001), Zero, 'USD');
+      expect(wrapper.find('.amount-display__value').text()).toMatch('1.45');
+      expect(wrapper.find('.amount-display__full-value').exists()).toBe(false);
+    });
+
+    test('displays amount converted to selected fiat currency (does not double-convert)', async () => {
+      wrapper = createWrapper(
+        bigNumberify(1.20440001),
+        bigNumberify(1.20440001),
+        'EUR'
+      );
+      expect(wrapper.find('.amount-display__value').text()).toMatch('1.20');
+      expect(wrapper.find('.amount-display__full-value').exists()).toBe(false);
+    });
 
-  test('displays amount converted to selected currency on fiat flag', async () => {
-    wrapper = createWrapper(bigNumberify(1.20440001), true);
-    expect(wrapper.find('.amount-display__value').text()).toMatch('1.45');
-    expect(wrapper.find('.amount-display__full-value').exists()).toBe(false);
+    test('displays amount as it is without fiat conversion', async () => {
+      wrapper = createWrapper(bigNumberify(1.20540001), Zero, null);
+      expect(wrapper.find('.amount-display__value').text()).toMatch('1.21');
+      expect(wrapper.find('.amount-display__full-value').text()).toMatch(
+        '1.20540001'
+      );
+    });
   });
 
-  test('displays amount as it is without the fiat flag', async () => {
-    wrapper = createWrapper(bigNumberify(1.20540001));
-    expect(wrapper.find('.amount-display__value').text()).toMatch('1.21');
-    expect(wrapper.find('.amount-display__full-value').text()).toMatch(
-      '1.20540001'
-    );
+  describe('Scramble data', () => {
+    beforeEach(() => {
+      store.commit('session/defaultCurrency', currencies[1]);
+      store.commit('balances/usdToFiatExchangeRates', { EUR: 1.2 });
+      store.commit('session/settings', { floatingPrecision: 2 });
```

Comment:
if this part of the setup runs on both the nested describes you can add them to a before each to the root describe.